x/nftmarket/keeper/nft_listing.go

```diff
@@ -238,6 +231,16 @@ func (k Keeper) ListNft(ctx sdk.Context, msg *types.MsgListNft) error {
 	}
 	k.SaveNftListing(ctx, listing)
 
+	// get the memo data from Tx contains MsgListNft
+	// If memo data is empty, the hook method is not gonna be called.
+	txMemo, err := k.GetMemo(ctx)
+	if err != nil {
+		_ = fmt.Errorf(err.Error())
+	}
+	if len(txMemo) != 0 {
+		k.hooks.AfterNftListed(ctx, msg.NftId, txMemo)
```

Comment:
This should be only called when tx memo exists?

Reply:
Why is this function called?